### ISSUE TITLE
[Release] Check list for v0.4.0

### ISSUE THREAD

**Issue by fidencio** [confidential-containers/community#74]
# v0.4.0

## Code freeze

- [x] - 1. Update image-rs to use the latest commit from ocicrypt-rs
        
        * https://github.com/confidential-containers/image-rs/blob/main/Cargo.toml
          * Change the revision
          * Run `cargo update -p ocicrypt-rs`

- [x] - 2. Update image-rs to use the latest commit from attestation-agent

        * https://github.com/confidential-containers/image-rs/blob/main/Cargo.toml
          * Change the revision
          * Run `cargo update -p image-rs`

- [x] - 3. Update Enclave CC to use the latest commit from image-rs

        * https://github.com/confidential-containers/enclave-cc/blob/main/src/enclave-agent/Cargo.toml
          * Change the revision
          * Run `cargo -p update image-rs`
        Note that you can point to your own fork here, so you don't actually do changes in the other projects
        before making sure this step works as expected.

- [x] - 4. Update Kata Containers to use the latest commit from image-rs

        * https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/Cargo.toml
          * Change the revision
          * Run `cargo -p update image-rs`
        Note that you can point to your own fork here, so you don't actually do changes in the other projects
        before making sure this step works as expected.

- [x] - 5. Update Kata Containers to use the latest attestation-agent

        * https://github.com/kata-containers/kata-containers/blob/CCv0/versions.yaml
          * Change the version

- [x] - 6. Update Kata Containers to use the latest td-shim

        * https://github.com/kata-containers/kata-containers/blob/CCv0/versions.yaml
          * Change the version

- [x] - 7. Check if there are new changes in the pre install payload script

        * https://github.com/confidential-containers/operator/tree/main/install/pre-install-payload
          * The last commit there must match what's in the following files as preInstall / postUninstall image
            * Enclave CC: https://github.com/confidential-containers/operator/blob/main/config/samples/enclave-cc/base/ccruntime-enclave-cc.yaml
            * Kata Containers:
              Note that for Kata Containers, we're looking for the newTag, below the quay.io/confidential-containers/container-engine-for-cc-payload image
              * s390x: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/s390x/kustomization.yaml
              * x86_64: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/default/kustomization.yaml

- [ ] - 8. Ensure the Operator is using the latest CI builds and that the Operator tests are passsing

        * Enclave CC:
          * SIM: https://github.com/confidential-containers/operator/blob/main/config/samples/enclave-cc/sim/kustomization.yaml
          * HW: https://github.com/confidential-containers/operator/blob/main/config/samples/enclave-cc/base/ccruntime-enclave-cc.yaml
          * Note that we need the quay.io/confidential-containers/runtime-payload-ci registry and enclave-cc-{SIM,HW}-latest tags
        * Kata Containers:
          * s390x: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/s390x/kustomization.yaml
          * x86_64: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/default/kustomization.yaml
          Note that we need the quay.io/confidential-containers/runtime-payload-ci registry and kata-containers-latest tag

- [x] - 9. Cut an ocicrypt-rs v0.4.0 release, if changes happened in the project

- [x] - 10. Cut an attestation-agent v0.4.0, if changes happened in the project

- [x] - 11. Cut an image-rs v0.4.0 release, using the latest release of:

        * ocicrypt-rs (redo step 1, but now using v0.4.0
        * attestation-agent (redo step 2, but now using v0.4.0

- [x] - 12. Cut a td-shim v0.4.0 release, if changes happened in the project

- [x] - 13. Update Enclave CC to use the released version of image-rs

        * redo step 3, but now using v0.4.0

- [x] - 14. Update Kata Containers to the latest released version of:

        * image-rs (redo step 4, but now using the v0.4.0)
        * attestation-agent (redo step 5, but now using the v0.4.0)
        * td-shim (redo step 6, but now using the v0.4.0)

- [x] - 15. Update the operator to use the images generated from the latest commit of both Kata Containers and Enclave CC

        * redo step 8, but now targetting the latest payload image generated for Kata Containers and Enclave CC

- [x] - 16. Make sure all the operator tests are passing

- [x] - 17. Cut an Enclave CC release

- [x] - 18. Add a new Kata Containers tag

## Release

- [x] - 19. Update the operator to use the release tags coming from Enclave CC and Kata Containers

        * redo step 8, but now targetting thje latest release of the payload image generated for Kata Containers eand Enclave CC

- [x] - 20. Update the Operator version

        * https://github.com/confidential-containers/operator/blob/main/config/release/kustomization.yaml#L7

- [x] - 21. Cut an operator release

- [ ] - 22. Make sure to update the release notes

        * https://github.com/confidential-containers/documentation/tree/main/releases/v0.4.0.md
        
- [ ] - 23. Poke Jens Freimann (jfreiman@redhat.com) to update the release to the OperatorHub

**Comment by jepio** [comment]
Notes from morning meeting:
- image-rs HEAD currently broken, @Xynnn007 will paste latest working commit
- attestation-agent needs a PR that is not merged yet (@stevenhorsman can you paste PR link?)
- I will look at steps 6+7
- @fidencio working on steps 1-3

**Comment by Xynnn007** [comment]
attestation-agent: c6cdff6dcd4f4ca45fa0fb7df439057d2dba15ea
image-rs: (depends on attestation-agent mentioned, might work I think): 48bb221d8e76cb35549f2ffa626d2f687f959bd2

**Comment by stevenhorsman** [comment]
@Xynnn007 - just to double-check - if we are using `c6cdff` of the attestation-agent, I don't think that https://github.com/kata-containers/kata-containers/pull/6249 is mandatory (I think it is still a valid and helpful change) for this release as I think the rework to openssl/rust-crypto happened after this point?

**Comment by Xynnn007** [comment]
> I think the rework to openssl/rust-crypto happened after this point?

Yes, the openssl/rust-crypto happened after `c6cdff`. Sorry I didn't try https://github.com/kata-containers/kata-containers/pull/6249, but in time order I simply think it should be postponed

**Comment by stevenhorsman** [comment]
> > I think the rework to openssl/rust-crypto happened after this point?
> 
> Yes, the openssl/rust-crypto happened after `c6cdff`. Sorry I didn't try [kata-containers/kata-containers#6249](https://github.com/kata-containers/kata-containers/pull/6249), but in time order I simply think it should/could be postponed

Great - thanks for confirming. @jepio @fidencio - I guess that means that 4 & 5 are unblocked now, do you want me to pick them up, or is someone else on it?

**Comment by jepio** [comment]
I'll tackle 5 in the same PR as 6, you could go for 4.

**Comment by fidencio** [comment]
On enclave-cc we're using [948db858579be538724266a650e840e75e7e824d](https://github.com/confidential-containers/enclave-cc/pull/110/commits/7e1ded00a7d92f77c6c6bab325a895fcd3085841)

**Comment by jepio** [comment]
My CCv0 PR is here: https://github.com/kata-containers/kata-containers/pull/6360.

Still need to check the payload scripts.

**Comment by fidencio** [comment]
So, this is what we may be using:
* ocicrypt-rs: 453c699a50326c2f55c7a357e90b9da1535d7c26
  * that's the version used with the commit we'll use from image-rs)
* image-rs: 948db858579be538724266a650e840e75e7e824d

This is the PR that I have for Enclave CC: https://github.com/confidential-containers/enclave-cc/pull/110/commits/7e1ded00a7d92f77c6c6bab325a895fcd3085841

** Noite ** image-rs is using a different version of attestation-agent than the version tested by @jepio, and I think we should just ignore that for now.  The release should be cut based on the version that Kata Containers is using.

**Comment by fidencio** [comment]
> ** Noite ** image-rs is using a different version of attestation-agent than the version tested by @jepio, and I think we should just ignore that for now. The release should be cut based on the version that Kata Containers is using.

My comment was clearly dubious.  We should go ahead and bump it on Kata Containers instead of following the version that's on image-rs, sorry.

**Comment by fidencio** [comment]
https://github.com/confidential-containers/enclave-cc/pull/118 is the Enclave CC PR for the v0.4.0 deps bump.

**Comment by fidencio** [comment]
https://github.com/confidential-containers/operator/pull/167 cover steps 19 and 20.

**Comment by fidencio** [comment]
Item 21 is done: https://github.com/confidential-containers/operator/releases/tag/v0.4.0
